File: selectionSort.cpp
#include <bits/stdc++.h>
using namespace std;


// swap given two values passed by reference to address
void swap(int *x, int *y){
	int temp = *x;
	*x = *y;
	*y = temp;
}


// function to implement selection sort technique
void selectionSort(int arr[], int n){
	int i,j, min_index;

	for(i=0;i< n-1;i++){
		min_index = i;
        // find minimum element in the unsorted array         
		for(j=i+1;j < n;j++){
			if(arr[j] < arr[min_index])
				min_index = j;
		}
        // swap minimum element with element corresponding to outer loop
		swap(&arr[min_index], &arr[i]);
	}
}


// function to print an array
void printArray(int arr[], int n){
	for(int i=0;i<n;i++){
		cout << arr[i] << " ";
	}
	cout << endl;
}

int main(){
	int size;
	cout << "Enter the length of array: ";
	cin >> size;
	int arr[size];
	cout << "Enter elements of the array: ";
	for(int i=0;i< size;i++){
		cin >> arr[i];
	}
	selectionSort(arr,size);
	cout << "Sorted array : \n";
	printArray(arr,size);
	return 0;

}

File: Selection_Sort.java
import java.util.Scanner;

public class Selection_Sort {

    void solution(int[] array){

        for(int i=0; i<array.length-1; i++){
            int pos = i;
            for(int j=i+1; j<array.length; j++){
                if(array[j]<array[i]) {
                    pos = j;
                }
            }
            int temp = array[pos];
            array[pos] = array[i];
            array[i] = temp;
        }
    }

    void display(int[] array){

        for (int i=0; i<array.length; ++i){
            System.out.print(array[i]+" ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Selection_Sort sort = new Selection_Sort();
        Scanner sc = new Scanner(System.in);


        System.out.println("Enter size of array");
        int size = sc.nextInt();

        System.out.println("Enter elements");
        int[] array = new int[size];
        for(int i=0; i<size; i++){
            array[i] = sc.nextInt();
        }
        sort.solution(array);
        sort.display(array);
    }
}
